src/pyseekdb/client/client_base.py

```
@@ -3092,23 +3106,25 @@ def _normalize_vectors(raw_embeddings: Any) -> list[list[float]]:
-            mapped = field_mapping.get(field.lower(), field)
-            if mapped not in source_fields:
-                source_fields.append(mapped)
-
-        return source_fields if source_fields else ["document", "metadata", "embedding"]
+        """
+        Infer OceanBase GET_SQL `_source` allowlist from include.
+        """
+        if include is None:
+            include = ["documents", "metadatas"]
+        if not isinstance(include, list) or not all(isinstance(item, str) for item in include):
+            raise TypeError("include must be a List[str] or None")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)